compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -549,6 +632,8 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
         if (ctx.definitions.isTupleType(tp))
           signature(tp).map(_ => "_").mkString("(", ", ", ")")
+        else if (sym.showFullName == "scala.collection.immutable.List")
```

Comment:
We should never do string comparisons to determine the type of something. Instead something like `sym == defn.ListClass` should be used, if `ListClass` is not defined in `Definitions.scala` then it should be added.